Rules/CTF/Scripts/CTF_GiveSpawnItems.as

```diff
@@ -267,32 +230,101 @@ void onRender(CRules@ this)
+		if (this.isWarmup())
+		{
+			wood_amount = 300;
+			stone_amount = 100;
+		}
+		string text = getTranslatedString("Go to a builder shop or a respawn point {SWITCH}to get a resupply of {WOOD} wood and {STONE} stone.")
+			.replace("{SWITCH}", need_to_switch_string)
+			.replace("{WOOD}", "" + wood_amount)
+			.replace("{STONE}", "" + stone_amount);
+
+		Vec2f offset = Vec2f(20, 64);
+		float x = getScreenWidth() / 3 + offset.x;
+		float y = getScreenHeight() - offset.y;
+		
 		if (next_items > getGameTime())
 		{
-			string action = (b.getName() == "builder" ? "Go Build" : "Go Fight");
-			if (this.isWarmup())
-			{
-				action = "Prepare for Battle";
-			}
+			// color = SColor(255, 255, 55, 55);
+			color = SColor(255, 255, 55, 55);
+			
+			text = getTranslatedString("Next resupply of {WOOD} wood and {STONE} stone in {SEC}{TIMESUFFIX}.")
+				.replace("{SEC}", "" + secs)
+				.replace("{TIMESUFFIX}", getTranslatedString(units))
+				.replace("{WOOD}", "" + wood_amount)
+				.replace("{STONE}", "" + stone_amount);
+
+			x = getScreenWidth() / 2;
+			y = getScreenHeight() / 3 - 70.0f;
+		}
+		
+		GUI::DrawTextCentered(text, Vec2f(x, y), color);
+	}
+	
+	// TODO(hobey): maybe only draw the cooldown/helptext for archer if low on arrows?
+	propname = getCTFTimerPropertyNameArcher(p);
+	if (name == "archer" && this.exists(propname))
+	{
+		s32 next_items = this.get_s32(propname);
+		
+		GUI::SetFont("menu");
+		
+		u32 secs = ((next_items - 1 - getGameTime()) / getTicksASecond()) + 1;
+		string units = ((secs != 1) ? " seconds" : " second");
+		
+		SColor color = SColor(200, 135, 185, 45);
+		int wood_amount = 100;
+		int stone_amount = 30;
+		if (this.isWarmup())
+		{
+			wood_amount = 300;
+			stone_amount = 100;
+		}
+		string text = getTranslatedString("Go to an archer shop or a respawn point to get a resupply of 30 arrows.");
 
-			u32 secs = ((next_items - 1 - getGameTime()) / getTicksASecond()) + 1;
-			string units = ((secs != 1) ? " seconds" : " second");
-			GUI::SetFont("menu");
-			GUI::DrawTextCentered(getTranslatedString("Next resupply in {SEC}{TIMESUFFIX}, {ACTION}!")
-							.replace("{SEC}", "" + secs)
-							.replace("{TIMESUFFIX}", getTranslatedString(units))
-							.replace("{ACTION}", getTranslatedString(action)),
-			              Vec2f(getScreenWidth() / 2, getScreenHeight() / 3 - 70.0f + Maths::Sin(getGameTime() / 3.0f) * 5.0f),
-			              SColor(255, 255, 55, 55));
+		Vec2f offset = Vec2f(20, 96);
+		float x = getScreenWidth() / 3 + offset.x;
+		float y = getScreenHeight() - offset.y;
+		
+		if (next_items > getGameTime())
+		{
+			// color = SColor(255, 255, 55, 55);
+			color = SColor(255, 255, 55, 55);
+			
+			text = getTranslatedString("Next resupply of 30 arrows in {SEC}{TIMESUFFIX}.")
+				.replace("{SEC}", "" + secs)
+				.replace("{TIMESUFFIX}", getTranslatedString(units));
+
+			x = getScreenWidth() / 2;
+			y = getScreenHeight() / 3 - 16.0f;
 		}
+		
+		GUI::DrawTextCentered(text, Vec2f(x, y), color);
```

Comment:
this could be made shorter, you shouldnt need to copy paste the code on a per class basis

Reply:
I've done some refactoring for this, could you please give a look if it's fine now?